eox_tenant/signals.py

```diff
@@ -217,3 +222,114 @@ def clear_tenant(sender, request, **kwargs):  # pylint: disable=unused-argument
     Signal: django.core.signals.got_request_exception
     """
     pass
+
+
+def tenant_context_addition(sender, body, headers, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver used to add context to the async process about the tenant associated to the task that need to be done.
+
+    Dispatched before a celery task is published. Note that this is executed in the process sending the task.
+    See:
+        https://celery.readthedocs.io/en/latest/userguide/signals.html#before-task-publish
+    """
+    get_host_func = AsyncTaskHandler().get_host_from_task(sender)
+    headers['eox_tenant_sender'] = get_host_func(body)
+
+
+def start_async_tenant(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver that runs on the async process to update the settings accordingly to the tenant.
+    Dispatched before a task is executed.
+    See:
+       https://celery.readthedocs.io/en/latest/userguide/signals.html#task-prerun
+    """
+    headers = sender.request.get('headers') or {}
+    http_host = headers.get('eox_tenant_sender')
+    if not http_host:
+        LOG.warning("Could not find the host information for eox_tenant.signals. ")
+        siteid = base_settings.SITE_ID
+        site = Site.objects.get(id=siteid)
+        http_host = site.name
```

Comment:
Why is the default tenant that of the `base_settings.SITE_ID` ?
Sure, django requires a default Site ID, but that does not mean that we are better of setting the tenants to that when we can not find the correct setting.

I would say that instead we can use a bogus string (e.g `"<no_tenant_selected>"`) or None as the domain and let the reset_or_keep_settings function handle it.

This is my opinion at least, better to have no tenant activated than to select the incorrect tenant by default.